### DISCUSSION TITLE
Shouldn't Forms be threaten as routing mechanism?

### DISCUSSION THREAD

**Discussion by edygar** [solidjs/solid-router#309] [Ideas]
Forms are a navigation tool on the web as much as the links are, where the form[method=get] is very comparable to the links, as it reflects directly on the URL, the form[method=POST] could really just be a pushState with associated state. 

A bit of context: I come from using react-router/remix, with which you can now assign `loader` function and an `action` function for each route to query and mutate data respectively, whenever there's navigation, `loader` is called before the transition, and if it was a transition from a Form[method=post], `action` is called even before. Although the whole data strategy pattern they endorse is very useful, I admit is outside the routing boundaries, but I truly believe we could implement the interception and push/replace of the states bit in this library.

**Comment by ryansolid** [comment]
Yeah maybe you have some thoughts regarding this: https://github.com/solidjs/solid-router/discussions/306
